pkg/controllers/ragengine_controller.go

```diff
@@ -33,14 +53,368 @@ func NewRAGEngineReconciler(client client.Client, scheme *runtime.Scheme, log lo
 }
 
 func (c *RAGEngineReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	ragEngineObj := &kaitov1alpha1.RAGEngine{}
+	if err := c.Client.Get(ctx, req.NamespacedName, ragEngineObj); err != nil {
+		if !errors.IsNotFound(err) {
+			klog.ErrorS(err, "failed to get RAG Engine", "RAG Engine", req.Name)
+		}
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+
+	klog.InfoS("Reconciling", "RAG Engine", req.NamespacedName)
+
+	result, err := c.addRAGEngine(ctx, ragEngineObj)
+	if err != nil {
+		return result, err
+	}
+
+	return result, nil
+}
+
+func (c *RAGEngineReconciler) addRAGEngine(ctx context.Context, ragEngineObj *kaitov1alpha1.RAGEngine) (reconcile.Result, error) {
+	err := c.applyRAGEngineResource(ctx, ragEngineObj)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 	return reconcile.Result{}, nil
 }
 
+// applyRAGEngineResource applies RAGEngine resource spec.
+func (c *RAGEngineReconciler) applyRAGEngineResource(ctx context.Context, ragEngineObj *kaitov1alpha1.RAGEngine) error {
```

Comment:
the same logic as https://github.com/Azure/kaito/blob/main/pkg/controllers/workspace_controller.go#L374